codejam-canvas/script.js

```diff
@@ -0,0 +1,47 @@
+const canvas = document.querySelector('#canvas');
+const ctx = canvas.getContext('2d');
+let inputs = document.querySelectorAll('input[name="size"]');
+for (let i = 0; i < inputs.length; i++) {
+    inputs[i].addEventListener('change', function () {
+        ctx.clearRect(0, 0, canvas.width, canvas.height);
+        getArr(this.value);
+    });
+}
+
+function getArr(scale) {
+    let url = '';
+    if (scale == '4') {
+        url = 'https://raw.githubusercontent.com/rolling-scopes-school/tasks/master/tasks/stage-2/codejam-canvas/data/4x4.json';
+    } else if (scale == '32') {
+        url = 'https://raw.githubusercontent.com/rolling-scopes-school/tasks/master/tasks/stage-2/codejam-canvas/data/32x32.json';
+    } else {
+        let img = document.createElement('img');
+        img.crossOrigin = "Anonymous";
+        img.src = 'https://raw.githubusercontent.com/rolling-scopes-school/tasks/master/tasks/stage-2/codejam-canvas/data/image.png';
+        img.onload = function() {
+            ctx.drawImage(img, 0, 0, canvas.width, canvas.height);
+            return;
+        }
+    }
+    fetch(url)
+        .then(res => res.json())
+        .then(data => drawPixel(data, scale));
+}
+
+function drawPixel(color, scale) {
+    let imageData = ctx.getImageData(0, 0, canvas.width, canvas.height);
+    for (let i = 0, k = 0; i < color.length; i++) {
+        for (let m = 0; m < canvas.width / scale; m++) {
+            for (let j = 0; j < color[i].length; j++) {
+                if (/[A-F]/.test(color[i][j])) color[i][j] = [parseInt(color[i][j].slice(0, 2), 16), parseInt(color[i][j].slice(2, 4), 16), parseInt(color[i][j].slice(4, 6), 16)];
```

Comment:
it would be more handy to place calculations of color converting to separate function and use it here